rs/execution_environment/src/scheduler/round_schedule.rs

```diff
@@ -229,4 +231,77 @@ impl RoundSchedule {
 
         (canisters_partitioned_by_cores, canisters)
     }
+
+    /// Finishes canister execution on thread.
+    pub fn finish_canister_execution_on_thread(
+        canister: &mut CanisterState,
+        executed_priority_ids: &mut BTreeMap<CanisterId, AccumulatedPriority>,
+        round_id: ExecutionRound,
+        is_first_iteration: bool,
+        rank: usize,
+    ) {
+        let full_message_execution = match canister.next_execution() {
+            NextExecution::None => true,
+            NextExecution::StartNew => false,
+            // We just finished a full slice of executions.
+            NextExecution::ContinueLong | NextExecution::ContinueInstallCode => true,
```

Comment:
Actually, a clueless question: should `ContinueLong` actually always qualify as a "full round"?

I.e. do we only start executing a new message if there's a full slice worth of instructions left? Because it not, we might pause a message execution after only a few instructions.

(I believe the answer is that we do only start a new message execution if we can afford a full slice, so mostly just double-checking my understanding.)

Reply:
> should ContinueLong actually always qualify as a "full round"?
It always does qualify in the code, right?